File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict
def solution(clothes):
    cntDict = defaultdict(int)
    for clothe in clothes:
        cntDict[clothe[1]] +=1
    ans = 1
    for kind in cntDict.keys():
        ans *= (cntDict[kind]+1)
    ans -= 1
    # print(cntDict)
    return ans

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    static Set<Set<String>> coms;
    static Set<String> tmp;
    static Map<String,Integer> kindCnt;
    public int solution(String[][] clothes) {
        kindCnt = new HashMap<>();
        coms = new HashSet<>();
        tmp = new HashSet<>();
        //종류 카운트 증가
        for(int i=0;i<clothes.length;i++){
            kindCnt.put(clothes[i][1],kindCnt.getOrDefault(clothes[i][1],0)+1);
        }
        int ans = 1;
        int kindsCnt = kindCnt.keySet().size();
        for(String s:kindCnt.keySet()){
            ans *= (kindCnt.get(s)+1);
        }
        return ans-1;
    }
}